src/main/java/com/couchbase/client/core/endpoint/AbstractEndpoint.java

```diff
@@ -116,7 +124,7 @@
     /**
      * The netty bootstrap adapter.
      */
-    private final BootstrapAdapter bootstrap;
+    private BootstrapAdapter bootstrap;
```

Comment:
this needs to be volatile, because the reconnect callback can be called from other threads.

Reply:
I was thinking about that, but I lacked the understanding of the threading model used in `Endpoint`. I've changed to `volatile`.